vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/MenuItemBase.java

```diff
@@ -70,6 +70,12 @@ public MenuItemBase(C contextMenu) {
         getElement().addEventListener("click", e -> {
             if (checkable) {
                 setChecked(!isChecked());
+
+                if (isKeepOpen()) {
+                    getElement().executeJs(
+                            "this.toggleAttribute('menu-item-checked')",
+                            isChecked());
```

Comment:
This logic could be changed to toggle `menu-item-checked` state in the web component instead.
However, that would be confusing as the web component does not change `checked` state.

Reply:
Moved this logic to the connector as discussed with @vursen.